applications/HDF5Application/custom_io/hdf5_container_component_io.h

```diff
@@ -40,13 +40,15 @@ namespace HDF5
 ///@{
 
 /// A class for IO of element data in HDF5.
-template <typename TContainerType, typename TContainerItemType, typename... TComponents>
+template <class TContainerType, class TContainerDataIO, class... TComponents>
 class KRATOS_API(HDF5_APPLICATION) ContainerComponentIO
```

Comment:
This is probably the most important class in the entire application, and yet it's not documented. We need detailed docs here about what this class does.

```suggestion
class KRATOS_API(HDF5_APPLICATION) ContainerComponentIO
```